File: Traning2/02_hist_v1.cpp
#include <bits/stdc++.h>

using namespace std;


unsigned long long find_max_s(unsigned long long n){
    unsigned long long temp;
    vector <unsigned long long> A;
    for (unsigned long long i=0; i!=n; i++){
        cin >> temp;
        A.push_back(temp);
    }
    cin >> temp;
    unsigned long long maxS=0;
    for (unsigned long long start=0; start<n; start++){
        unsigned long long min_height = 0;
        for (unsigned long long end=start; end<n; end++){
            min_height = *min_element(A.begin()+start, A.begin() + end);
            unsigned long long S = min_height * (end-start+1);
            maxS = maxS > S ? maxS : S;
        }
    }
    // cout << maxS;
    return maxS;
}

int main(){
    unsigned long long n;
    vector<unsigned long long> res;
    while (true){
        cin >> n;
        if (n==0){
            break;
        }
        res.push_back(find_max_s(n));
        
    }
    // cout << 5;

    for (unsigned long long i; i!= res.size(); i++){
        cout << res[i] << endl;
    }

    return 0;
}

File: Training2/02_hist_v1.py
def find_max_S(A:list, n: int):
    max_S = 0

    for start in range(0, n, 1):

        for end in range(start, n, 1):
            min_height = min(A[start:end+1])
            S = min_height * (end - start +1)
            max_S = max(S, max_S)

    return max_S

res = []
while True:
    lines = input()
    numbers = [int(n) for n in lines.split() if n!= '']
    # print(numbers)
    if numbers[0] == 0:
        break

    n = numbers[0]
    A = numbers[1:]
    res.append(find_max_S(A, n))

for r in res:
    print(r)